### PR TITLE
pass default args when --pass-args is set

### LINKED ISSUE #3449: `--pass-args` should pass default `ARG`s too
consider

```
VERSION --pass-args 0.7
FROM alpine

a:
  ARG foo=alpha
  BUILD --pass-args +b

b:
  ARG foo=beta
  RUN --no-cache echo in b with $foo

test:
  ARG foo=hello
  BUILD --pass-args +a
```

When one runs `earthly +test`, it ultimately prints out `in b with beta`; which feels wrong.

Instead we expect the default value `ARG foo=hello` to be set, then passed via `BUILD --pass-args +a`, which ultimately should print out `in b with hello`.

### PR DESCRIPTION
**Pull request by alexcb**
fixes #3449

### CODE REVIEW

File: earthfile2llb/converter.go [hunk 1]
```diff
@@ -1828,7 +1828,9 @@ func (c *Converter) prepBuildTarget(ctx context.Context, fullTargetName string,
 	}
 	// Don't allow transitive overriding variables to cross project boundaries (unless --pass-args is used).
 	propagateBuildArgs := !relTarget.IsExternal()
-	if passArgs || propagateBuildArgs {
+	if passArgs {
+		overriding = variables.CombineScopes(overriding, c.varCollection.Overriding(), c.varCollection.Args(), c.varCollection.Globals())
+	} else if propagateBuildArgs || passArgs {
```
Comment (mikejholly) [inline earthfile2llb/converter.go:1833]: This condition seems a bit confusing as it won't be executed if `passArgs` is true given L1831.

File: variables/scope.go [hunk 2]
```diff
@@ -21,6 +22,22 @@ func NewScope() *Scope {
 	}
 }
 
+// DebugString returns a string that can be printed while debugging
+func (s *Scope) DebugString() string {
```
Comment (mikejholly) [inline variables/scope.go:26]: :+1:

**Review by mikejholly — APPROVED** [review]
LGTM.

File: earthfile2llb/converter.go (hunk 1 above)
Comment (alexcb) [reply earthfile2llb/converter.go:1833]: good catch! that slipped back in when I was rebasing.

**Review by mikejholly — APPROVED** [review]

**Pull request merged**